crates/loader/src/local.rs

```diff
@@ -88,7 +89,7 @@ impl LocalLoader {
 
         let variables = variables
             .into_iter()
-            .map(|(name, v)| Ok((name.to_string(), locked_variable(v)?)))
+            .map(|(name, v)| Self::env_checker((name.to_string(), locked_variable(v)?)))
```

Comment:
Doing this here means appears to mean we can't even build the LockedApp without having the variables available.  How does this worth with e.g. `spin registry push` or `spin cloud deploy`?